lib/Tomcat/ServletTest.pm

```diff
@@ -24,23 +24,43 @@ use constant TIMEOUT => 60;
 
 # test all Servlet examples
 sub test_all_examples() {
-    my ($self) = shift;
+    my ($self, $mod_jk) = @_;
 
     # array with example test function and number of tabs required to select the example
     # the two servlet 4.0 examples are skipped
-    my @servlet_examples = ([\&hello, 2], [\&request_information, 4], [\&request_header, 4], [\&request_parameters, 4], [\&cookies, 4], [\&sessions, 4], [\&async0, 3], [\&async1, 1], [\&async2, 1], [\&async3, 1], [\&stocksticker, 1], [\&byte_counter, is_sle('<12-sp4') ? 2 : 1], [\&number_writer, 1]);
+    my @servlet_examples = ([\&hello, 2], [\&request_information, 4], [\&request_header, 4], [\&request_parameters, 4], [\&cookies, 4], [\&sessions, 4]);
+    if (!defined $mod_jk) {
```

Comment:
Because that `my ($self, mod_jk) = @_;` above, you can totally get away with:
```suggestion
if(!$mod_jk){
```

And doing the same change for other places where you're doing a similar call